core/model/modx/modlexicon.class.php

```diff
@@ -457,12 +457,35 @@ private function _parse($str,$params) {
         if (!$str) return '';
         if (empty($params)) return $str;
 
+        $params = $this->_flatten($params);
         foreach ($params as $k => $v) {
             $str = str_replace('[[+'.$k.']]',$v,$str);
         }
         return $str;
     }
 
+    /**
+     * Flattens an array by dot notation
+     *
+     * @access private
+     * @param array $array The array to flatten
+     * @return array The processed array
+     */
+    private function _flatten($array) {
```

Comment:
It does not make sense to use underscore to mark a method as private when we have a special keyword in the language. It would be nice also to have configured symbols, like a dot, but as it only for lexicons and private, it is not required.

Reply:
I have adopted that underscore notation by the invoking method name _parse. We can add that flatten method to a core helper class, if that is needed. There we can make the separator configurable.

Reply:
It would be a nice feature for 3.x I guess, but for 2.x I am fine with the current solution.